### DISCUSSION TITLE
How does `Expr` receive literal string?

### DISCUSSION THREAD

**Discussion by TD-Sky** [TedDriggs/darling#286] [Q&A] [answered]
I'm writing a attribute macro used on struct:

```rust
#[mc(key = "this is a literal string")]
struct Foo;
```

The receiver type:

```rust
#[derive(Debug, FromMeta)]
struct Args {
    value: syn::Expr,
}
```

The `value` expects a literal string or `syn::Path` to a value. Parsing it `NestedMeta::parse_meta_list` and `FromMeta::from_list` will get error `Unknown literal value `this is a literal string`. Are there any solutions?

**Comment by TedDriggs** [comment, marked as answer]
This is likely running into #229, so you should use [`util::parse_expr::preserve_str_literal`](https://docs.rs/darling/latest/darling/util/parse_expr/fn.preserve_str_literal.html) to handle it.
